src/DocExplorer/components/DocExplorer.tsx

```diff
@@ -50,21 +55,21 @@ export const DocExplorer: React.FC = () => {
 
   const [showSidebar, setShowSidebar] = useState(true);
 
-  const { selectedDoc, selectDoc, selectedDocUrl } = useSelectedDoc({
+  const [selectedDocLink, setSelectedDocLink] = useSelectedDocLink({
     rootFolderDoc,
     changeRootFolderDoc,
   });
 
-  const selectedDocLink = rootFolderDoc?.docs.find(
-    (doc) => doc.url === selectedDocUrl
-  );
+  const [selectedDoc] = useDocument(selectedDocLink?.url);
```

Comment:
curious what the purpose of pulling this out was!

Reply:
I like to keep hooks simple. useSelectedDocLink doesn't use the doc and it's straight forward to call useDocument in the actual component.